x-pack/plugins/gis/public/components/widget_overlay/attribution_control/view.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+import _ from 'lodash';
+import {
+  EuiText,
+  EuiPanel,
+  EuiLink,
+} from '@elastic/eui';
+
+
+export class AttributionControl  extends React.Component {
+
+  constructor() {
+    super();
+    this.state = {
+      uniqueAttributions: []
+    };
+  }
+
+  componentDidMount() {
+    this._isMouunted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMouunted = false;
+  }
+
+  componentDidUpdate() {
+    this._syncMbMapWithAttribution();
+  }
+
+  _syncMbMapWithAttribution = async () => {
+
+    const attributionPromises = this.props.layerList.map(layer => {
+      return layer.getAttributions();
+    });
+    const attributions = await Promise.all(attributionPromises);
+    if (!this._isMouunted) {
+      return;
+    }
+
+    const uniqueAttributions = [];
+    for (let i = 0; i < attributions.length; i++) {
```

Comment:
Is there a reason why you are using for loops instead of forEach?

Reply:
yes, didn't want to create too many nested lexical scopes. it keeps the code a little flatter, both when reading it and at run-time (smaller callstack, less heap allocation).